tests/testthat/math_token_spacing/non_strict_math_spacing_all-out.R

```diff
@@ -1 +1 @@
-1 + +1   -       3 / 23 * 3 ^ 4
+1 + +1   -       3 / 23 * 3 ^4
```

Comment:
I have no idea why this happens:

``` r
styler::style_text("1 - 3 * 4 ^ 5", strict = FALSE)
#> 1 - 3 * 4 ^ 5
```

Reply:
That's actually what I'd expect because `strict = FALSE` means no strict implementation, right? Or do you prefer to set the spaces to exactly zero around `^` even if `strict = FALSE`?

Reply:
I think the reason this might seem inconsistent with the other test is that there is no space around `^` in any of the other tests, but I think that is because in *-in.R files, we always have `x^k`(so no space in between) and hence the tests actually don't test whether `x ^ k` is turned into `x^k`.